# What's New in the ANNOVAR software


: 2015Feb17: The ExAC03 that has indel left-realigned is now available in ANNOVAR. It includes all populations and seven subpopulations (AFR,AMR,EAS,FIN,NFE,OTH,SAS). (Update 20150218: file was updated due to a user-reported bug, please download again). (Update 20150302: multiple users reported that chrX contains only ALL population, so I updated the file again, and please download again. Thanks for the bug report)

as: 2015Jan06: I wrote a new article with some thoughts and guidelines on processing VCF files and assigning dbSNP identifiers to variants. You can take a look by clicking the "VCF Processing" menu to the left.

as: 2014Dec22: I now created new dbSNP that has indel left-realigned and provide to ANNOVAR users, including avsnp138 in hg19 coordiante and avsnp142 in hg19 and hg38 coordinate. Please read "VCF Processing" menu to the left.

as: 2014Dec16: Updated 1000g2014oct are available now that addressed some issues with indel mismatch. Please read "VCF Processing" menu to the left. In addition, esp6500siv2_all, esp6500siv2_aa, esp6500siv2_ea are available, and both hg19 and hg38 are available to download now.

as: 2014Nov12: ANNOVAR new version is available, with significantly reduce memory usage for filter annotation, improved compatibility for unconventional chromosome names for species such as tomato, fixed a problem in exon numbering for splice variants in reverse strand. Registered users will receive an email with link to download.


